skycatalogs/objects/snana_object.py

```diff
@@ -25,6 +26,10 @@ def __init__(self, ra, dec, id, object_type, belongs_to, belongs_index,
     def _get_sed(self, mjd=None):
         if mjd is None:
             mjd = self._belongs_to._mjd
+        if mjd is None:
```

Comment:
Since you're re-implementing `get_observer_sed_component` in this class, I'd put this check there and not in a private function.  I'd also suggest implementing an `__init__` for `SnanaCollection` and have it check there if `mjd` is being set rather than waiting to check it here.  I can't see any use case where we'd load an `SnanaCollection` without providing an `mjd` so that should be checked at the outset, making the separate test here unnecessary.

Reply:
It seems unlikely  but it not impossible that someone would want to do it.  What if they just want to see the distribution of SNe in some region? What percentage of galaxies have an associated SN?

Reply:
If they wanted that info, I would just suggest that they access the parquet file directly.  Doing so through skyCatalogs seems unnecessarily unwieldy when they could just read in the data with `pandas` or similar.

Reply:
I'm moving the check to `get_observer_sed_component` as suggested.

Reply:
> If they wanted that info, I would just suggest that they access the parquet file directly. 
skyCatalogs could still do region selection for them.
I think it depends on what they're doing and what they have been doing.  If they're set up to use skyCatalogs already that might be easier.  I'd like to leave the option open.